pkg/controller/provider/container/vsphere/model.go

```diff
@@ -811,24 +811,33 @@ func (v *VmAdapter) Apply(u types.ObjectUpdate) {
 				}
 			case fGuestDisk:
 				if disks, cast := p.Val.(types.ArrayOfGuestDiskInfo); cast {
+					var diskMountPoints []model.DiskMountPoint
 					for _, info := range disks.GuestDiskInfo {
 						// Default to 0 so the policy can flag missing mappings
 						guestDiskKey := int32(0)
 						if len(info.Mappings) > 0 {
 							// VMware guarantees at least one mapping when non-empty
 							guestDiskKey = info.Mappings[0].Key
 						}
-
-						newGuestDisk := model.GuestDisk{
+						diskMountPoint := model.DiskMountPoint{
 							DiskPath:       info.DiskPath,
 							Capacity:       info.Capacity,
 							FreeSpace:      info.FreeSpace,
 							FilesystemType: info.FilesystemType,
 							Key:            guestDiskKey,
 						}
 
-						v.updateOrAppendGuestDisk(newGuestDisk)
+						// Check for m.model.Disks with the same key (disk keys are expected to be unique)
+						for i, disk := range v.model.Disks {
+							if disk.Key == diskMountPoint.Key {
```

Comment:
NOTE: this logic will set the disk/PVC win-drive-letter to use the first diskMountPoint.DiskPath that match this disk/PVC key

it's a valid way of finding the drive letter in a multiple mount points case